neqo-transport/src/streams.rs

```diff
@@ -0,0 +1,464 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Stream ID and stream index handling.
+
+use crate::fc::{
+    LocalStreamsFlowControls, ReceiverFlowControl, RemoteStreamLimits, SenderFlowControl,
+};
+use crate::frame::Frame;
+use crate::packet::PacketBuilder;
+use crate::recovery::RecoveryToken;
+use crate::recv_stream::{RecvStream, RecvStreams};
+use crate::send_stream::{SendStream, SendStreams, TransmissionPriority};
+use crate::stats::FrameStats;
+use crate::stream_id::{StreamId, StreamType};
+use crate::tparams::{self, TransportParametersHandler};
+use crate::ConnectionEvents;
+use crate::{Error, Res};
+use neqo_common::{qtrace, qwarn, Role};
+use std::cell::RefCell;
+use std::rc::Rc;
+
+pub struct Streams {
+    role: Role,
+    tps: Rc<RefCell<TransportParametersHandler>>,
+    events: ConnectionEvents,
+    flow_control_sender: Rc<RefCell<SenderFlowControl<()>>>,
+    flow_control_receiver: Rc<RefCell<ReceiverFlowControl<()>>>,
+    remote_streams_fc: RemoteStreamLimits,
+    local_streams_fc: LocalStreamsFlowControls,
+    pub(crate) send_streams: SendStreams,
+    pub(crate) recv_streams: RecvStreams,
```

Comment:
This I'm less sure of, but it seems like it avoids some redundancy.

```suggestion
    pub(crate) send: SendStreams,
    pub(crate) recv: RecvStreams,
```